packages/client/src/components/SafeSettings/components/Owners/ReviewAddSafeOwner.js

```diff
@@ -0,0 +1,66 @@
+import { useContext } from "react";
+import { useModalContext } from "contexts";
+import { useHistory } from "react-router-dom";
+import { Web3Context } from "contexts/Web3";
+import { formatAddress } from "utils";
+import OwnerCard from "./OwnerCard";
+
+const ReviewAddSafeOwner = ({
+  owner,
+  newThreshold,
+  safeOwners,
+  onBack,
+  treasury
+}) => {
+  const history = useHistory();
+  const { closeModal } = useModalContext();
+  const { address } = treasury
+  const { proposeAddSigner, setTreasury } =
+  useContext(Web3Context);
+
+  const renderRow = (label, value) => {
+    return (
+      <div className="columns m-0 py-1  border-light-top">
+        <label className="column px-0 has-text-grey">{label}</label>
+        <div className="column ml-2 px-0 has-text-black has-text-weight-bold">{value}</div>
+      </div>
+    );
+  };
+
+  const onSubmitClick = async () => {
+    await proposeAddSigner(formatAddress(owner.address), newThreshold);
+    setTreasury(address, {
+      safeOwners: [...safeOwners, { ...owner, verified: false }],
+    });
+    closeModal();
+    history.push(`/safe/${address}`);
+  }
+
+  return (
+    <>
+      <div className="p-4">
+        <h2 className="is-size-4 has-text-black">Review Updates</h2>
+      </div>
```

Comment:
We can remove this, and add it as a modal config, i'll write a comment where exactly